app/services/integration/kubernetes_service.rb

```diff
@@ -279,26 +378,40 @@ def update_resource(existing, resource)
   end
 
   def persist_proxy(entry)
-    routes = build_proxy_routes(entry)
-
-    cleanup_routes persist_resources(routes)
+    if use_openshift_route?
```

Comment:
I'm wondering if we could create two separate subclasses: one for OpenShift service, and another one for plain Kubernetes, and override the methods where the `if use_openshift_route?` check is currently done.
Then maybe we could select the correct service in DiscoverIntegrationService...

Reply:
I agree with @mayorova.
Adding this `use_openshift_route` option forced us to add `if`s on multiple methods that that now have to have a different behaviour depending on this new option.
Rather than doing that, since there are multiple changes to be done, we could define a new class with all the behaviour we expect when we talk about OpenShift routes and leave this one to care about Kubernetes.
With this, we'd have different points in code responsible for each scenario.